app/scripts/controllers/network/network-controller.ts

```diff
@@ -839,14 +839,34 @@ export class NetworkController extends EventEmitter {
    * supports EIP-1559; otherwise it doesn't.
    *
    * @param provider - A provider, which is guaranteed to be available.
+   * @param maxRetries - The maximum number of retries to attempt fetching the latest block. Defaults to 3.
    * @returns A promise that resolves to true if the network supports EIP-1559
-   * and false otherwise.
+   * and false otherwise. Throws an error if unable to fetch the last block.
    */
   async #determineEIP1559Compatibility(
     provider: SafeEventEmitterProvider,
+    maxRetries = 3,
```

Comment:
As this is a private method so only called once, is it clearer to move this to a file level constant?